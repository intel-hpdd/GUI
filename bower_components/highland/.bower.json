{
  "name": "highland",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "version": "2.7.1",
  "_release": "2.7.1",
  "_resolution": {
    "type": "version",
    "tag": "2.7.1",
    "commit": "197822ee483f1e38fd410863d0edb4ad07d06d4d"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.7.1",
  "_originalSource": "highland"
}