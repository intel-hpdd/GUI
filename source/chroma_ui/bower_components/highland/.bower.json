{
  "name": "highland",
  "version": "2.4.0",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "_release": "2.4.0",
  "_resolution": {
    "type": "version",
    "tag": "2.4.0",
    "commit": "7ff0deb4abec773009324072d8855f394cc93467"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.4.0",
  "_originalSource": "highland"
}