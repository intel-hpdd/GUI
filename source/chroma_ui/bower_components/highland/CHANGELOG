Changelog
=========

This file does not aim to be comprehensive (you have git history for that),
rather it lists changes that might impact your own code as a consumer of
this library.

2.0.0
-----

* The `source.merge()` algorithm now evaluates the entire source stream before
  reading from all of the resulting streams in parallel (previously it would
  start reading as soon as the source emitted the next stream)
* The `merge()` function now attempts to balance inputs more fairly. For example,
  if stream A has 100 values buffered and stream B gets a new value after 100ms,
  if we read at 200ms we'll get a value from each stream. Previously it would
  exhaust the stream A buffer before reading from stream B.
