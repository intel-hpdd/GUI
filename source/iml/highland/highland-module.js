// @flow

//
// Copyright (c) 2018 DDN. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import angular from 'angular';
import highland from 'highland';

export default angular
  .module('highland', [])
  .value('Î»', highland)
  .value('highland', highland).name;
