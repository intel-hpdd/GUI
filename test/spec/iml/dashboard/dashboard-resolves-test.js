import highland from 'highland';

import {
  mock,
  resetAll
} from '../../../system-mock.js';

describe('dashboard resolves', () => {
  let s, spy, store, broadcaster, mod;

  beforeEachAsync(async function () {
    spy = jasmine.createSpy('spy');
    s = highland();

    broadcaster = jasmine.createSpy('broadcaster')
      .and
      .callFake(x => () => x);

    store = {
      select: jasmine
        .createSpy('select')
        .and
        .returnValue(s)
    };

    mod = await mock('source/iml/dashboard/dashboard-resolves.js', {
      'source/iml/store/get-store.js': {
        default: store
      },
      'source/iml/broadcaster.js': {
        default: broadcaster
      }
    });
  });

  afterEach(resetAll);

  describe('fs stream', () => {
    let fsStream;

    beforeEach(() => {
      fsStream = mod.dashboardFsB()();
    });

    it('should be a broadcaster', () => {
      expect(broadcaster)
        .toHaveBeenCalledOnce();
    });

    it('should select from the store', () => {
      expect(store.select)
        .toHaveBeenCalledOnceWith('fileSystems');
    });

    it('should stream data', () => {
      s.write(['foo']);

      fsStream.each(spy);

      expect(spy)
        .toHaveBeenCalledOnceWith(['foo']);
    });
  });

  describe('host stream', () => {
    let hostStream;

    beforeEach(() => {
      hostStream = mod.dashboardHostB()();
    });

    it('should be a broadcaster', () => {
      expect(broadcaster)
        .toHaveBeenCalledOnce();
    });

    it('should select from the store', () => {
      expect(store.select)
        .toHaveBeenCalledOnceWith('server');
    });


    it('should stream data', () => {
      s.write(['foo']);

      hostStream.each(spy);

      expect(spy)
        .toHaveBeenCalledOnceWith(['foo']);
    });
  });

  describe('target stream', () => {
    let targetStream;

    beforeEach(() => {
      targetStream = mod.dashboardTargetB()();
    });

    it('should be a broadcaster', () => {
      expect(broadcaster)
        .toHaveBeenCalledOnce();
    });

    it('should select from the store', () => {
      expect(store.select)
        .toHaveBeenCalledOnceWith('targets');
    });


    it('should stream data', () => {
      s.write(['foo']);

      targetStream.each(spy);

      expect(spy)
        .toHaveBeenCalledOnceWith(['foo']);
    });
  });
});
