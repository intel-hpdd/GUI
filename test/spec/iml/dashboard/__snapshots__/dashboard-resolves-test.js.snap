// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard resolves fs stream should join the data 1`] = `
Array [
  Object {
    "bytes_free": 3841384448,
    "bytes_total": 6058418176,
    "client_count": 5,
    "files_free": 2621169,
    "files_total": 2621440,
    "id": 1,
  },
  Object {
    "bytes_free": 1753678638,
    "bytes_total": 4163518176,
    "client_count": 2,
    "files_free": 3225169,
    "files_total": 7251340,
    "id": 2,
  },
]
`;
